src/relay/backend/graph_plan_memory.cc

```diff
@@ -106,40 +109,43 @@ class StorageAllocaBaseVisitor : public ExprVisitor {
   virtual void CreateToken(const ExprNode* op, bool can_realloc) = 0;
 };
 
-
 class StorageAllocaInit : protected StorageAllocaBaseVisitor {
  public:
   explicit StorageAllocaInit(common::Arena* arena)
       : arena_(arena) {}
 
-
   /*! \return The internal token map */
   std::unordered_map<const ExprNode*, std::vector<StorageToken*> >
   GetInitTokenMap(const Function& func) {
+    node_device_map_ = CollectDeviceInfo(func);
     this->Run(func);
     return std::move(token_map_);
   }
 
-
  protected:
   using StorageAllocaBaseVisitor::VisitExpr_;
 
   void CreateToken(const ExprNode* op, bool can_realloc)  final {
     CHECK(!token_map_.count(op));
     std::vector<StorageToken*> tokens;
+    int device_id = node_device_map_.count(GetRef<Expr>(op))
+                        ? node_device_map_[GetRef<Expr>(op)]->value
+                        : 0;
```

Comment:
shall we add 0 as a placeholder in `DLDeviceType` so that others will not use it for other special purpose by mistake.
also `device_id` -> `device_type`

Reply:
@yzhliu Yes, I also thought that it's probably necessary to have a field in DLDeviceType, like 'kDLUNDEFINED = 0'. Let's keep this first. I will send a RFC later to hear from more people because it needs a slight change in dlpack.